s3tests_boto3/functional/test_s3.py

```diff
@@ -12133,3 +12138,601 @@ def test_object_lock_uploading_obj():
     eq(response['ObjectLockLegalHoldStatus'], 'ON')
     client.put_object_legal_hold(Bucket=bucket_name, Key=key, LegalHold={'Status':'OFF'})
     client.delete_object(Bucket=bucket_name, Key=key, VersionId=response['VersionId'], BypassGovernanceRetention=True)
+
+
+def _wait_replication_time_az():
+    time.sleep(2)
```

Comment:
i'd really like to find a way to get these tests running in teuthology, but sleeps like this tend to cause a lot of timing failures in the sepia lab environment; performance there is highly variable because we co-locate lots of ceph daemons on the same machine, and run most of them under valgrind

waiting for replication to finish is a problem that we've (mostly) solved in the rgw multisite test suite with the help of radosgw-admin commands. you can find some notes on the test suite in https://github.com/ceph/ceph/blob/master/src/test/rgw/test_multi.md. the zonegroup_meta_checkpoint() and zonegroup_bucket_checkpoint() functions are how we wait for metadata and bucket sync to complete before verifying the results

would you be willing to port these test cases there? @yuvalif did an excellent job adding coverage for the pubsub zone in https://github.com/ceph/ceph/pull/26299, which could serve as an example for the archive zone

@alimaredia raised a separate concern about s3-tests, which aspires to be a generic test suite for all s3 implementations, so we'd like to avoid too many tests that only apply to ceph/rgw

Reply:
@jmunhoz just out of curiosity how were you orchestrating ceph locally while you were developing these tests?

Reply:
@cbodley @alimaredia thanks for reviewing and sharing pointers to test_multi and pubsub patches. I uploaded the new code at https://github.com/ceph/ceph/pull/29359

Reply:
> @jmunhoz just out of curiosity how were you orchestrating ceph locally while you were developing these tests?

@alimaredia I set up one only cluster (MON=1 MDS=0 MGR=1 RGW=2 OSD=3) with 1 realm, 1 zonegroup and 2 zones. All of them running on localhost with endpoints on 8001 and 8002 ports. One of the zones is an archive zone.

MON=1 MDS=0 MGR=1 RGW=2 OSD=3

Reply:
@jmunhoz note that by using the multi_site framework, it would be easy to use the same code for testing setups with more than 2 zones